### PR TITLE
ESB mediator for Machine Learner

### PR DESCRIPTION
**Pull request by manoramahp**

### CODE REVIEW

File: components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.machinelearner.ui/pom.xml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon</groupId>
+            <artifactId>org.wso2.carbon.mediator.service</artifactId>
+            <version>4.2.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon</groupId>
+            <artifactId>org.wso2.carbon.logging</artifactId>
+            <version>4.3.0</version>
```
Comment (chanakaudaya) [inline components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.machinelearner.ui/pom.xml]: Please move the version numbers from this module to the parent pom of the repository. Most of these references would be in the parent pom. No need to add the version number here. Please do the same for the remaining dependencies.

File: components/mediators/machine-learner/org.wso2.carbon.mediator.machinelearner/pom.xml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.synapse</groupId>
+            <artifactId>synapse-core</artifactId>
+            <version>2.1.3-wso2v1-SNAPSHOT</version>
```
Comment (chanakaudaya) [inline components/mediators/machine-learner/org.wso2.carbon.mediator.machinelearner/pom.xml]: Here also, please remove the version numbers from sub modules and move them to the parent pom of the repository.

**Comment by manoramahp** [comment]
Included the above mentioned suggestions.

Note : 
Please check whether the relative path given for `<propertiesFile>` is correct in the mediator features pom.xml files. (For bam mediator features, smooks mediator features, etc.)

`<propertiesFile>../../../etc/feature.properties</propertiesFile>`

File: pom.xml [hunk 3]
```diff
@@ -2043,6 +2080,10 @@
         <org.apache.commons.logging.version>1.1.1</org.apache.commons.logging.version>
         <project.scm.id>wso2-scm-server</project.scm.id>
 
+        <spark.core.version>1.2.1.wso2v2</spark.core.version>
+        <ml.core.version>1.0.0-SNAPSHOT</ml.core.version>
+        <ml.commons.version>1.0.0-SNAPSHOT</ml.commons.version>
+
```
Comment (isudana) [inline pom.xml]: We need to depend on the released version of these components

**Comment by manoramahp** [comment]
Updated the pom.xml with the released versions of org.wso2.carbon.ml components.

**Comment by manoramahp** [comment]
Closing the PR as this is merged into carbon-ml repo
https://github.com/manoramahp/carbon-ml/tree/master/components/extensions

**Pull request closed without merging**